crates/cubecl-core/src/frontend/element/atomic.rs

```diff
@@ -0,0 +1,375 @@
+use super::{
+    init_expand_element, ExpandElementBaseInit, ExpandElementTyped, LaunchArgExpand, Numeric,
+    Vectorized, I32, I64,
+};
+use crate::{
+    frontend::{CubeContext, CubePrimitive, CubeType, ExpandElement, UInt},
+    ir::{
+        BinaryOperator, CompareAndSwapOperator, Elem, IntKind, Item, Operator, UnaryOperator,
+        Vectorization,
+    },
+    prelude::KernelBuilder,
+    unexpanded,
+};
+
+pub trait Atomic: Sized + CubeType
```

Comment:
I would add some documentation before merging.

Reply:
Good idea - I was changing things so much I forgot to go back and add docs. I'll go add it to all the new public types/traits.